bundles/communities-ugc-migration/src/main/java/com/adobe/communities/ugc/migration/importer/ScoresSimpleImportServlet.java

```diff
@@ -0,0 +1,125 @@
+
+        if (fileRequestParameters != null && fileRequestParameters.length > 0 
+            && !fileRequestParameters[0].isFormField() 
+            && fileRequestParameters[0].getFileName().endsWith(".json")) {
+                
+            final InputStream inputStream = fileRequestParameters[0].getInputStream();
+            final JsonParser jsonParser = new JsonFactory().createParser(inputStream);
+            JsonToken token = jsonParser.nextToken(); // get the first token
+            
+            if (token.equals(JsonToken.START_OBJECT)) {
+                try {
+                    final UserPropertiesManager userManager = userPropertiesService.createUserPropertiesManager(resolver);
+                    readJsonUserScore(jsonParser, resolver, resourcePath);
+                } catch (RepositoryException e) {
+                    throw new ServletException("Unable to communicate with Jcr repository", e);
+                } catch (NullPointerException ex) {
+                    throw new ServletException("Please enter the path to the communities page in the url using query parameter 'path'");
+                }
+            } else {
+                throw new ServletException("Expected a start object token, got " + token);
+            }
+        } else {
+            log.info("Invalid file");
+        }
+
+    }
+
+
+    private void readJsonUserScore(final JsonParser jsonParser, ResourceResolver resolver, String resourcePath)
+            throws ServletException, IOException {
+
+        JsonToken jsonToken = jsonParser.nextToken();
+
+        Resource componentResource = resolver.getResource(resourcePath); // CRX node path to the communities site passed via the URL
+        Resource scoreRuleResource = resolver.getResource(resourcePath + "/jcr:content"); // Child node to the communities page where scoringRules are applied
+//        ValueMap valueMap = scoreRuleResource.adaptTo(ValueMap.class);
+//        String scoringRule = valueMap.get("scoringRules", ""); //scoringRules property value being fetched
+
+        while (!jsonToken.equals(JsonToken.END_OBJECT)) {
+            String authId = jsonParser.getCurrentName();
+            jsonToken = jsonParser.nextToken();
+            Long score = jsonParser.getValueAsLong();
+
+            try {
+                scoringService.saveScore(resolver, authId, componentResource, scoreRuleResource, score);
+                log.info("auth_id:" + authId + "& Score:" + score);
+                log.info("Component Resource:" + componentResource);
+                log.info("Score Rule Resource:" + scoreRuleResource);
+//                log.info("Scoring Rule:" + scoringRule);
+
+                jsonToken = jsonParser.nextToken();
+            } catch (RepositoryException e) {
+                e.printStackTrace();
+            } catch (NullPointerException ex) {
+                throw new ServletException("Please enter the path to the communities page in the url using query parameter 'path'");
```

Comment:
null pointer exception is for no query parameter passed in the URL i.e., 'path' parameter pointing to the communities page.